### PR TITLE
Element hiding: inject strict hide rules as style tag if enabled by privacy config

### PR DESCRIPTION
**Pull request by dharb**
**Asana link:** https://app.asana.com/0/1203557590179903/1203556051691984/f
**Corresponding privacy config PR:** https://github.com/duckduckgo/privacy-configuration/pull/619

**Description**
This PR adds the ability to inject a style tag for strict hide rules. This can be toggled on/off globally and for specific domains using privacy config. If toggled off, element hiding will work exactly as it did prior to this change.

### CODE REVIEW

File: src/features/element-hiding.js [hunk 1]
```diff
@@ -157,51 +158,83 @@ function isDomNodeEmpty (node) {
...
+    rules.forEach((rule, i) => {
+        if (i === rules.length - 1) {
+            styleTag = styleTag.concat(rule.selector, '{')
+        } else {
+            styleTag = styleTag.concat(rule.selector, ',')
+        }
+    })
+
+    styleTag = styleTag.concat('display:none!important;min-height:0!important;height:0!important;}</style>')
+    document.head.insertAdjacentHTML('beforeend', styleTag)
```
Comment (jonathanKingston) [inline src/features/element-hiding.js:235]: This is risking XSS through using a variant of innerHTML; instead we should create the style element, innerText the rules and then append the node.

**Review by jonathanKingston — CHANGES_REQUESTED** [review]
We should fix the potential XSS risk here (even if from our own remote). 

Are we not going to allow for remotely disabling the style tag injection?

**Comment by dharb** [comment]
> Are we not going to allow for remotely disabling the style tag injection?

No we are - see [here](https://github.com/duckduckgo/content-scope-scripts/pull/229/files#diff-fe23b6c57bd95dc1c110287c02534dc14b910516ac0dd67fce7610ae7ff8a628R280-R287) and [here](https://github.com/duckduckgo/content-scope-scripts/pull/229/files#diff-fe23b6c57bd95dc1c110287c02534dc14b910516ac0dd67fce7610ae7ff8a628R198-R200), as well as the corresponding privacy config changes in https://github.com/duckduckgo/privacy-configuration/pull/619. We can disable globally or by domain.

I'll address the styletag potential xss issue you flagged now.

File: src/features/element-hiding.js (hunk 1 above)
Comment (dharb) [reply src/features/element-hiding.js]: Good catch, updated to address.

**Review by jonathanKingston — DISMISSED** [review]
Looks great to me thanks for making this configurable! :)

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**